--- a/RxCocoa/iOS/DataSources/RxCollectionViewReactiveArrayDataSource.swift
+++ b/RxCocoa/iOS/DataSources/RxCollectionViewReactiveArrayDataSource.swift
@@ -43,9 +43,6 @@
     , RxCollectionViewDataSourceType {
     typealias Element = S
 
-    override init(cellFactory: @escaping CellFactory) {
-        super.init(cellFactory: cellFactory)
-    }
     
     func collectionView(_ collectionView: UICollectionView, observedEvent: Event<S>) {
         Binder(self) { collectionViewDataSource, sectionModels in
